repo2docker/buildpacks/base.py

```diff
@@ -307,6 +326,8 @@ def compose_with(self, other):
         build_script_files.update(other.build_script_files)
         result.build_script_files = build_script_files
 
+        # FIXME: Validate that they all have same base image!
+        result.base_distro = self.base_distro
```

Comment:
Do we need to validate that they are the same? If they should all be the same, we can use a class attribute instead of allowing subclasses to deviate if they shouldn't be allowed to do so.

Reply:
I'm thinking of #191 before getting this PR merged maybe